cmake/modules/FindQatDrv.cmake

```diff
@@ -0,0 +1,113 @@
+# - Find_QatDrv
+# Find the qat driver library and includes
+#
+# QatDrv_INCLUDE_DIRS = where to find cap.h, qae_mem.h, etc
+# QatDrv_LIBRARIES - List of libraries when using qat driver
+# QatDrv_FOUND - True if qat driver found
+
+set(expected_version "v1.7.L.4.14.0-00031")
+set(expected_version_url "https://www.intel.com/content/www/us/en/download/19081/intel-quickassist-technology-intel-qat-driver-for-linux-for-intel-server-boards-and-systems-based-on-intel-62x-chipset.html")
+
+function(check_qatdrv_version expected_version expected_version_url)
+  if(NOT DEFINED ENV{ICP_ROOT})
+    set(ENV{ICP_ROOT} "/opt/APP/driver/QAT")
+  endif()
+  find_path(QATDRV_Dir 
+	    name versionfile
+	    HINTS $ENV{ICP_ROOT}
```

Comment:
`$ENV{ICP_ROOT}` should be a hint set by user, not a regular internal variable used in this way. you could just put:
```cmake
function(get_qatdrv_version qat_include_dir)
  # ...
  set(QatDrv_VERSION_MAJOR ${QAT_VERSION3} PARENT_SCOPE)
  set(QatDrv_VERSION_MINOR ${QAT_VERSION4} PARENT_SCOPE)
  set(QatDrv_VERSION_PATCH ${QAT_VERSION5} PARENT_SCOPE)  
  set(QatDrv_VERSION_TWEAK ${QAT_VERSION6} PARENT_SCOPE)  
  set(QatDrv_VERSION_COUNT 4 PARENT_SCOPE)
  set(QatDrv_VERSION ${QatDrv_VERSION_MAJOR}.${QatDrv_VERSION_MINOR}.${QatDrv_VERSION_PATCH}.${QatDrv_VERSION_TWEAK})
endfunction()

find_path(QATDRV_INCLUDE_DIR
  name quickassist/include/cpa.h
  HINTS $ENV{ICP_ROOT} /opt/APP/driver/QAT
  NO_DEFAULT_PATH)
if(QATDRV_INCLUDE_DIR)
  get_qatdrv_version(${QATDRV_INCLUDE_DIR}/versionfile)
  set(QatDrv_INCLUDE_DIRS
    # ...
    ${QATDRV_INCLUDE_DIR}/quickassist/include)
endif()
foreach(component ${QatDrv_FIND_COMPONENTS})
  # ...
endforeach()

include(FindPackageHandleStandardArgs)
find_package_handle_standard_args(QatDrv
  REQUIRED_VARS QatDrv_LIBRARIES QatDrv_INCLUDE_DIRS
  VERSION_VAR QatDrv_VERSION)
if(NOT QatDrv_FOUND)
  # inform user where to download QAT driver package
endif()
```